### PR TITLE
bpo-17232: Clarify docs for -O and -OO command line options

### PR DESCRIPTION
**Pull request by csabella**
Original patch by Terry Reedy.

https://bugs.python.org/issue17232

### CODE REVIEW

File: Modules/main.c [hunk 1]
```diff
@@ -96,8 +96,10 @@ static const char usage_2[] = "\
          if stdin does not appear to be a terminal; also PYTHONINSPECT=x\n\
 -I     : isolate Python from the user's environment (implies -E and -s)\n\
 -m mod : run library module as a script (terminates option list)\n\
--O     : optimize generated bytecode slightly; also PYTHONOPTIMIZE=x\n\
--OO    : remove doc-strings in addition to the -O optimizations\n\
+-O     : remove assert and __debug__-dependent statements; add .opt1 before\n\
+         .pyc extension; also PYTHONOPTIMIZE=x\n\
+-OO    : do -O changes and also discard docstrings; add .opt2 before\n\
```
Comment (vadmium) [inline Modules/main.c]: Why do these say “.opt1” etc without a dash, but the other cases have a dash in “.opt-1” etc?

Comment (terryjreedy) [reply Modules/main.c]: I presume oversight on Cheryl's part.  I will fix.

File: Doc/using/cmdline.rst [hunk 2]
```diff
@@ -260,12 +260,23 @@ Miscellaneous options
 
 .. cmdoption:: -O
 
-   Turn on basic optimizations.  See also :envvar:`PYTHONOPTIMIZE`.
+   Remove assert statements and any code conditional on the value of
+   :const:`__debug__`.  Change the filename extension for compiled
+   (:term:`bytecode`) files to include ``.opt-1`` before the ``.pyc``
```
Comment (terryjreedy) [inline Doc/using/cmdline.rst]: I will change these to read 'Augment the filename for compiled bytecode files by adding .opt-1 before the .pyc extension. '  It does not matter whether one thinks of the addition as a 2nd extension or not, and the change is strictly an addition.

File: Modules/main.c (hunk 1 above)
Comment (terryjreedy) [reply Modules/main.c]: I presume oversight.  I checked my cache and '-' is correct.

**Review by terryjreedy — APPROVED** [review]
Thanks for moving this along.  I will edit and test locally.
Approve was a click mistake.

File: Misc/python.man [hunk 3]
```diff
@@ -164,10 +164,13 @@ for the named module and runs the corresponding
...
 .TP
 .B \-O
-Turn on basic optimizations.  Given twice, causes docstrings to be discarded.
+Remove assert statements and any code conditional on the value of
+__debug__; augment the filename for compiled (bytecode) files by
+adding .opt-1 before the .pyc extension.
 .TP
 .B \-OO
-Discard docstrings in addition to the \fB-O\fP optimizations.
+Do \fB-O\fP and also discard docstrings; change the filename for
```
Comment (terryjreedy) [inline Misc/python.man:172]: Cheryl, can you explain the \fB, \fP markup and verify that you somehow tested that it does 'the right thing'?  I looked at both the web page and python -h output.

Comment (csabella) [reply Misc/python.man:172]: When I changed the line, I kept that part from the original line and simply add the other text around it.  However, it appears the Github does attempt to show what will happen with it.  If you click on the 'View' button for this source, Github applies the markup in non-raw mode.  This shows up as bolded as does some of the text in the `-i` and `-I` option that has similar (though not exactly the same) markup.

File: Modules/main.c (hunk 1 above)
Comment (csabella) [reply Modules/main.c]: Yes, thank you for noticing that, Martin, and thank you, Terry, for fixing it.  It was an oversight.

File: Misc/python.man (hunk 3 above)
Comment (terryjreedy) [reply Misc/python.man:172]: Thanks.  Enough to comfortably merge without personal test.

**Pull request merged**